src/agency/agency.repository.ts

```diff
@@ -0,0 +1,19 @@
+import { Inject } from "@nestjs/common";
+import { DB, DbType } from "src/global/providers/db.provider";
+import { DataRetrievalException } from "src/exception";
+import { FindManyRepo } from "./agency.repository.schema";
+
+export class AgencyRepository {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  async findMany(): Promise<FindManyRepo | undefined> {
```

Comment:
The reference borough code is incorrect in a subtle way. The 'findMany' method return an array. If nothing met the criteria, it will return an empty array. We should actually expect this to always be defined. (This also goes for the boroughs endpoint. But that's out of scope for this ticket. We'll get back to the boroughs endpoint one day.) .

```suggestion
  async findMany(): Promise<FindManyRepo> {
```